File: mobile/lib/features/financial/domain/model/payway_callback_log.dart
class PayWayCallbackLog {
  final String id;
  final String userId;
  final String transactionId;
  final String merchantRefNo;
  final String rawPayloadJson;
  final bool processed;
  final String? processingError;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const PayWayCallbackLog({
    required this.id,
    required this.userId,
    required this.transactionId,
    required this.merchantRefNo,
    required this.rawPayloadJson,
    required this.processed,
    required this.processingError,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: backend/src/main/java/com/lifeos/backend/financial/infrastructure/provider/payway/log/PayWayCallbackLog.java
package com.lifeos.backend.financial.infrastructure.provider.payway.log;

import com.lifeos.backend.common.base.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Entity
@Table(name = "payway_callback_logs")
@Getter
@Setter
public class PayWayCallbackLog extends BaseEntity {
    private UUID userId;
    private String transactionId;
    private String merchantRefNo;

    @Column(columnDefinition = "TEXT")
    private String rawPayloadJson;

    private Boolean processed;

    @Column(columnDefinition = "TEXT")
    private String processingError;
}